test/Ocelot.AcceptanceTests/RoutingTests.cs

```diff
@@ -808,6 +810,44 @@ public void should_return_response_200_with_placeholder_for_final_url_path()
                 .BDDfy();
         }
 
+        [Theory]
+        [InlineData("/downstream/test/{testId}", "/upstream/test/{testId}", "/upstream/test/1", "/downstream/test/1")]
+        [InlineData("/downstream/test/{testId}", "/upstream/test/{testId}", "/upstream/test/", "/downstream/test/")]
+        [InlineData("/downstream/test/{testId}", "/upstream/test/{testId}", "/upstream/test", "/downstream/test")]
+        [InlineData("/downstream/test/{testId}", "/upstream/test/{testId}", "/upstream/test123", null)]
+        [InlineData("/downstream/{version}/test/{url}", "/upstream/{version}/test/{url}", "/upstream/v1/test/123", "/downstream/v1/test/123")]
+        [InlineData("/downstream/{version}/test", "/upstream/{version}/test", "/upstream/v1/test", "/downstream/v1/test")]
+        [InlineData("/downstream/{version}/test", "/upstream/{version}/test", "/upstream/test", null)]

```

Comment:
😮 I wonder why `string downstreamURL` is `null` here? 😵 
Should not it be `"/downstream/test"` because of empty `{version}` placeholder?

Reply:
is null because there is no match in here because the placeholder is in the middle, that is a requirement on the issue